src/primitives/transaction.h

```diff
@@ -61,13 +61,36 @@ class CTxIn
     CScript scriptSig;
     uint32_t nSequence;
 
+    /* Setting nSequence to this value for every input in a transaction
+     * disables nLockTime. */
+    static const uint32_t SEQUENCE_FINAL = 0xffffffff;
+    /* If this flag set, CTxIn::nSequence is NOT interpreted as a
+     * relative lock-time. Setting the most significant bit of a
+     * sequence number disabled relative lock-time. */
+    static const uint32_t SEQUENCE_LOCKTIME_DISABLED_FLAG = (1 << 31);
+    /* If CTxIn::nSequence encodes a relative lock-time and this flag
+     * is set, the relative lock-time has units of 512 seconds,
+     * otherwise it specifies blocks with a granularity of 1. */
+    static const uint32_t SEQUENCE_LOCKTIME_SECONDS_FLAG = (1 << 22);
```

Comment:
Bikeshedding, I think `SEQUENCE_LOCKTIME_SECONDS_FLAG` is not a good name because this flag determines the type of relative lock-time, i.e. timebased or block based. I wonder if `SEQUENCE_LOCKTIME_TYPE_FLAG` would be better.